strawberryfields/ops.py

```diff
@@ -1105,13 +1105,13 @@ class MZgate(Gate):
             (R(\phi_{ex})\otimes I)
 
     Args:
-        phi_ex (float): external phase
         phi_in (float): internal phase
+        phi_ex (float): external phase
     """
     ns = 2
 
-    def __init__(self, phi_ex, phi_in):
-        super().__init__([phi_ex, phi_in])
+    def __init__(self, phi_in, phi_ex):
+        super().__init__([phi_in, phi_ex])
 
     def _decompose(self, reg, **kwargs):
         # into local phase shifts and two 50-50 beamsplitters
```

Comment:
I think we also need to adjust this method, also swapping the indices of the parameters used for the various phase shifts.

Reply:
```

    def _decompose(self, reg, **kwargs):
        # into local phase shifts and two 50-50 beamsplitters
        return [
            Command(Rgate(self.p[1]), reg[0]),
            Command(BSgate(np.pi/4, np.pi/2), reg),
            Command(Rgate(self.p[0]), reg[0]),
            Command(BSgate(np.pi/4, np.pi/2), reg)
        ]
```
right?

Reply:
Isn't it enough to only make the swap in one place (either the order of the arguments or where p[0] and p[1] appear)?

Reply:
`p[0]` is the first argument (=now internal phase), `p[1]` the second (=now external phase). The command sequence in my snippet above should first apply the external phase, then a BS, then the internal phase, then another BS. So it needs to be changed to what I proposed above, no?

Reply:
Right!

Reply:
And the math/docstring on line 1103!

Reply:
Good catch!